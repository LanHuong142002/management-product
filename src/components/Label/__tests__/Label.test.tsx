import { cleanup, render } from '@testing-library/react';
import { Label, LabelProps } from '@components';

describe('Testing Label component', () => {
  beforeEach(() => {
    cleanup();
  });

  const defaultProps = {
    text: '200',
    variant: 'primary',
  } as LabelProps;

  it('Should render Label component', () => {
    const { container } = render(<Label {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Testing props variant primary have class', () => {
    const { getByText } = render(<Label {...defaultProps} />);

    const text = getByText('200');

    expect(text).toBeInTheDocument();
  });

  it('Should render Label component without variant', () => {
    const { container } = render(<Label text='200' />);

    expect(container).toMatchSnapshot();
  });
});
