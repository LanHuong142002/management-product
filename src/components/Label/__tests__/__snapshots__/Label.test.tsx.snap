// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Label component Should render Label component 1`] = `
<div>
  <div
    class="label-wrapper label-primary"
  >
    <span>
      200
    </span>
  </div>
</div>
`;

exports[`Testing Label component Should render Label component without variant 1`] = `
<div>
  <div
    class="label-wrapper "
  >
    <span>
      200
    </span>
  </div>
</div>
`;
